File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_233604/typescript/008_prime_numbers/solution.ts
const N = parseInt(process.argv[2]);

if (isNaN(N) || N < 2) {
  process.exit(0);
}

// Sieve of Eratosthenes
const sieve = new Array<boolean>(N + 1).fill(true);
sieve[0] = false;
sieve[1] = false;

for (let i = 2; i * i <= N; i++) {
  if (sieve[i]) {
    for (let j = i * i; j <= N; j += i) {
      sieve[j] = false;
    }
  }
}

const lines: string[] = [];
for (let i = 2; i <= N; i++) {
  if (sieve[i]) {
    lines.push(i.toString());
  }
}

if (lines.length > 0) {
  process.stdout.write(lines.join("\n") + "\n");
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_233604/python/008_prime_numbers/solution.py
import sys

def main():
    if len(sys.argv) < 2:
        return
    N = int(sys.argv[1])

    if N < 2:
        return

    # Sieve of Eratosthenes
    sieve = [True] * (N + 1)
    sieve[0] = False
    sieve[1] = False
    for i in range(2, int(N ** 0.5) + 1):
        if sieve[i]:
            for j in range(i * i, N + 1, i):
                sieve[j] = False

    for i in range(2, N + 1):
        if sieve[i]:
            print(i)

main()